periphery/collect_periphery_info.bzl

```diff
@@ -0,0 +1,64 @@
+load("@build_bazel_rules_swift//swift:providers.bzl", "SwiftInfo")
+
+PeripheryInfo = provider(
+    doc = "Provides indexstore information for a target's recursive dependencies.",
+    fields = {
+        "periphery_file_target_mapping": "A File listing the sources of a target and paths to their respective indexstores.",
+        "periphery_indexstore": "A File representing the indexstore of a target.",
+        "srcs": "A list of sources for a target.",
+    },
+)
+
+def _collect_periphery_info_aspect_imp(target, ctx):
+    periphery_file_target_mapping = []
+    periphery_indexstore = []
+    srcs = []
+
+    # Only act on SwiftInfo targets that are not testonly, and exist in the current workspace
```

Comment:
In v3, the idea is that we can toggle the inclusion of test targets in the analysis based on the `--exclude-tests` option. So would should be able to ignore `testonly` here and instead list these test targets: https://github.com/peripheryapp/periphery/blob/v3/Sources/PeripheryKit/Generic/GenericProjectDriver.swift#L10

Reply:
Sounds great, should we keep this implementation intact for now and when that API is ready we could switch on that to filter targets